### PR TITLE
Tex command: more helpful error messages

### LINKED ISSUE #415: Tex command: more helpful error messages
Most LaTeX rendering errors look like this right now, which contains the error message returned by the rTeX API:

Discord embed, with message: pdflatex exited with non-zero return code

This is not as helpful as it could be. For the most part, users are getting this because they forgot to use `$` to get into math mode. We should probably replace the API error message with our own tip.

### PR DESCRIPTION
**Pull request by thxo**
### Description

Changes:
- Use a custom error message for LaTeX rendering errors, instead of using the message from the API.
- Add some LaTeX environment info to the docstring.

Fixes #415

### Type of change

- Bug fix 

### How has this been tested?

Manual testing.

### Checklist

- [x] Issue exists for PR
- [ ] Code reviewed by the author
- [x] Code documented (comments or other documentation)
- [x] Changes tested
- [x] All tests pass (see [`DEVELOPING.md`][0], if it exists)
- [x] [`CHANGELOG.md`][1] updated if needed
- [x] Informative commit messages
- [x] Descriptive PR title

[0]: /DEVELOPING.md
[1]: /CHANGELOG.md

### CODE REVIEW

File: exts/tex.py [hunk 1]
```diff
@@ -28,7 +28,10 @@ def __init__(self, bot: commands.Bot):
 
     @commands.command(name="tex", aliases=["latex"], category=cmn.Cats.UTILS)
     async def tex(self, ctx: commands.Context, *, expr: str):
-        """Renders a LaTeX expression."""
+        """Renders a LaTeX expression.
+
+        In document mode by default. To render math, add `$` around math expressions.
```
Comment (classabbyamp) [inline exts/tex.py:33]: is it called document mode or text mode or what?

File: exts/tex.py [hunk 2]
```diff
@@ -45,7 +48,11 @@ async def tex(self, ctx: commands.Context, *, expr: str):
                 if render_result["status"] != "success":
                     embed = cmn.embed_factory(ctx)
                     embed.title = "LaTeX Rendering Failed!"
-                    embed.description = render_result.get("description", "Unknown error")
+                    embed.description = ("Here are some common reasons:\n"
+                                         "- Did you forget to use math mode? Surround math expressions with `$`,"
+                                         " like `$x^3$`.\n"
+                                         "- Are you using a command from a package? It might not be available.\n"
+                                         "- Are you including the document headers? We already did that for you.")
```
Comment (classabbyamp) [inline exts/tex.py:55]: what if we used unicode bullet point ("•") for these?

File: exts/tex.py [hunk 3]
```diff
@@ -45,7 +48,11 @@ async def tex(self, ctx: commands.Context, *, expr: str):
                 if render_result["status"] != "success":
                     embed = cmn.embed_factory(ctx)
                     embed.title = "LaTeX Rendering Failed!"
-                    embed.description = render_result.get("description", "Unknown error")
+                    embed.description = ("Here are some common reasons:\n"
```
Comment (classabbyamp) [inline exts/tex.py:51]:
```suggestion
                    embed.description = ("Here are some possible reasons for failure:\n"
```
maybe?

**Review by 0x5c — DISMISSED** [review]

File: exts/tex.py (hunk 3 above)
Comment (0x5c) [reply exts/tex.py:51]: ¯\\_(ツ)\_/¯

**Review by classabbyamp — APPROVED** [review]

**Pull request merged**